### PR TITLE
Remove ModelViewer

### PR DESCRIPTION
**Pull request by edu-gp**
- Delete ModelViewer, move its logic into either db.model::Model or db.model::FileInference.
- Wire up the UI so we can still view the`index` and `show` pages for Tasks.

Jing this sets you up well to rewire the `create` and `edit` pages for Tasks.

### CODE REVIEW

File: backend/tasks.py [hunk 1]
```diff
@@ -213,27 +212,31 @@ def train(id):
...
+
+        # Write it to a temp file and send it.
+        import tempfile
+        import os
+        from werkzeug.utils import secure_filename
         from flask import send_file
-        return send_file(final_fname, mimetype='text/csv', cache_timeout=0, as_attachment=True)
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            name = f"{secure_filename(task.name)}__{stem(fname)}.csv"
```
Comment (JingJZ160) [inline backend/tasks.py:233]: Since we are accessing `.name` of a `task`, should we add a check on whether `task` is none?

File: db/task.py [hunk 2]
```diff
@@ -211,21 +211,24 @@ def parse_jinjafied(field, value):
...
-        if mv is not None:
-            return NLPModel(self.task_id, mv.version)
-        else:
-            return None
+        raise Exception("Deprecated Function")
+        # mv = self.get_active_model_viewer()
+        # if mv is not None:
+        #     return NLPModel(self.task_id, mv.version)
+        # else:
+        #     return None
```
Comment (JingJZ160) [inline db/task.py:234]: Do we still need those comments?

File: shared/utils.py [hunk 3]
```diff
@@ -89,3 +89,20 @@ class PrettyDefaultDict(defaultdict):
...
 def gen_uuid():
     return str(uuid.uuid4())
+
+
+def file_len(fname):
+    try:
+        with open(fname) as f:
+            for i, l in enumerate(f):
+                pass
+        return i + 1
```
Comment (JingJZ160) [inline shared/utils.py:99]: Wait...variable `i` is still visible outside the `for` loop in Python?

Comment (edu-gp) [reply shared/utils.py:99]: Yep

**Pull request merged**